master/internal/api_config_policies.go

```diff
@@ -67,10 +67,36 @@ func (*apiServer) GetWorkspaceConfigPolicies(
 	ctx context.Context, req *apiv1.GetWorkspaceConfigPoliciesRequest,
 ) (*apiv1.GetWorkspaceConfigPoliciesResponse, error) {
 	if !configpolicy.ValidWorkloadType(req.WorkloadType) {
-		return nil, fmt.Errorf("invalid workload type: %s", req.WorkloadType)
+		errMessage := fmt.Sprintf("invalid workload type: %s.", req.WorkloadType)
+		if len(req.WorkloadType) == 0 {
+			errMessage = noWorkloadErr
+		}
+		return nil, status.Errorf(codes.InvalidArgument, errMessage)
 	}
-	data, err := stubData()
-	return &apiv1.GetWorkspaceConfigPoliciesResponse{ConfigPolicies: data}, err
+
+	resp := apiv1.GetWorkspaceConfigPoliciesResponse{}
+	configPolicies, err := configpolicy.GetTaskConfigPolicies(
+		ctx, ptrs.Ptr(int(req.WorkspaceId)), req.WorkloadType)
+	if err != nil {
+		return nil, err
+	}
+	policyMap := map[string]interface{}{}
+	if configPolicies.InvariantConfig != nil {
+		var configMap map[string]interface{}
+		if err := yaml.Unmarshal([]byte(*configPolicies.InvariantConfig), &configMap); err != nil {
+			return nil, fmt.Errorf("unable to unmarshal json: %w", err)
+		}
+		policyMap["invariant_config"] = configMap
+	}
+	if configPolicies.Constraints != nil {
+		var constraintsMap map[string]interface{}
+		if err := yaml.Unmarshal([]byte(*configPolicies.Constraints), &constraintsMap); err != nil {
+			return nil, fmt.Errorf("unable to unmarshal json: %w", err)
+		}
+		policyMap["constraints"] = constraintsMap
+	}
+	resp.ConfigPolicies = configpolicy.MarshalConfigPolicy(policyMap)
```

Comment:
Maybe we can initialize `resp` closer to its first use case
```suggestion
	resp := apiv1.GetWorkspaceConfigPoliciesResponse{ConfigPolicies: configpolicy.MarshalConfigPolicy(policyMap)}
```